lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1305,6 +1305,16 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
           if (isVaried(cai->getSrc(), i))
             recursivelySetVariedIfDifferentiable(cai->getDest(), i);
         }
+        else if (auto *sei = dyn_cast<StructExtractInst>(&inst)) {
```

Comment:
You'll also want to check whether `CotangentVector` is `@_fieldwiseProductSpace`, because if it's not, we consider `@noDerivative` as having had no effect on conformance synthesis, so it could still have a VJP we would call when it's active.

It would be nice to move the code that checks for `@_fieldwiseProductSpace` into a helper.